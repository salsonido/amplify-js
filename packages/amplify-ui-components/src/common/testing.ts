import {
	boolean as booleanKnob,
	select as selectKnob,
	text as textKnob,
} from '@storybook/addon-knobs';

export const pixelThreshold = 10;

export const selectors = {
	// amplify-form-field
	formFieldDescription: '[data-test=form-field-description]',
};

export const knobs = {
	labelKnob(labelPlaceholder: string) {
		return textKnob('Label', labelPlaceholder);
	},

	descriptionKnob(descriptionPlaceholder: string) {
		return textKnob('Description', descriptionPlaceholder);
	},

	hintKnob(hintPlaceholder: string) {
		return textKnob('Hint', hintPlaceholder);
	},

	inputTypeKnob(inputTypePlaceholder: string) {
		return selectKnob(
			'Type',
			[
				'date',
				'email',
				'number',
				'password',
				'search',
				'tel',
				'text',
				'url',
				'time',
			],
			inputTypePlaceholder
		);
	},

	placeholderKnob(placeholderPlaceholder: string) {
		return textKnob('Placeholder', placeholderPlaceholder);
	},

	buttonTypeKnob(buttonTypePlaceholder: string) {
		return selectKnob(
			'Button type',
			['button', 'submit', 'reset'],
			buttonTypePlaceholder
		);
	},

	buttonTextKnob(buttonTextPlaceholder: string) {
		return textKnob('Button text', buttonTextPlaceholder);
	},

	toggleKnob(toggleDefault: boolean) {
		return booleanKnob('Toggle', toggleDefault);
	},

	disabledKnob(disabledDefault: boolean) {
		return booleanKnob('Disabled', disabledDefault);
	},

	formSectionHeaderKnob(headerPlaceholder: string) {
		return textKnob('Header text', headerPlaceholder);
	},

	formSectionSubmitKnob(submitPlaceholder: string) {
		return textKnob('Submit button text', submitPlaceholder);
	},

	tooltipAutoShowKnob(autoShowDefault: boolean) {
		return booleanKnob('Tooltip autoshow', autoShowDefault);
	},
};
