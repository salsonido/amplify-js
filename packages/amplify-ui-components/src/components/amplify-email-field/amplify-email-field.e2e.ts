import { newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-email-field e2e:', () => {
	it('should render a amplify-email-field', async () => {
		const page = await newE2EPage();
		await page.setContent(`<amplify-email-field></amplify-email-field>`);
		const el = await page.find('amplify-email-field');
		expect(el).not.toBeNull();

		const screenshot = await page.compareScreenshot('amplify-email-field', {
			fullPage: true,
		});
		expect(screenshot).toMatchScreenshot({
			allowableMismatchedPixels: pixelThreshold,
		});
	});
});
