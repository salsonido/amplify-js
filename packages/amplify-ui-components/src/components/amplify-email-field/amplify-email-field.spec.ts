import { newSpecPage } from '@stencil/core/testing';
import { AmplifyEmailField } from './amplify-email-field';

describe('amplify-email-field spec:', () => {
	describe('Component logic ->', () => {
		let emailField;

		beforeEach(() => {
			emailField = new AmplifyEmailField();
		});

		it('should render `email` for the field id', () => {
			expect(emailField.fieldId).toEqual('email');
		});

		it('should have `label` render to `Email Address *` by default', () => {
			expect(emailField.label).toEqual('Email Address *');
		});

		it('should have `placeholder` render to `Enter your email address` by default', () => {
			expect(emailField.placeholder).toEqual('Enter your email address');
		});

		it('should have `required` set to `false` by default', () => {
			expect(emailField.required).toBe(false);
		});
	});
	describe('Render logic ->', () => {
		it('should render `placeholder` and `label values', async () => {
			const page = await newSpecPage({
				components: [AmplifyEmailField],
				html: `<amplify-email-field label="New email *" placeholder="Enter new email address"></amplify-email-field>`,
			});

			expect(page.root).toMatchSnapshot();
		});

		it('should render default values by default', async () => {
			const page = await newSpecPage({
				components: [AmplifyEmailField],
				html: `<amplify-email-field></amplify-email-field>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});
