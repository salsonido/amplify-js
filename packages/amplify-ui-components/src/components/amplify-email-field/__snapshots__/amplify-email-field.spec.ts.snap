// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-email-field spec: Render logic -> should render \`placeholder\` and \`label values 1`] = `
<amplify-email-field label="New email *" placeholder="Enter new email address">
  <amplify-form-field fieldid="email" label="New email *" name="email" placeholder="Enter new email address" type="email"></amplify-form-field>
</amplify-email-field>
`;

exports[`amplify-email-field spec: Render logic -> should render default values by default 1`] = `
<amplify-email-field>
  <amplify-form-field fieldid="email" label="Email Address *" name="email" placeholder="Enter your email address" type="email"></amplify-form-field>
</amplify-email-field>
`;
